pkg/build/fuchsia.go

```diff
@@ -52,22 +52,24 @@ func (fu fuchsia) build(params *Params) error {
 		return err
 	}
 
-	// Fuchsia images no longer include ssh keys. Manually append the ssh public key to the zbi.
-	sshZBI := filepath.Join(params.KernelDir, "out", arch, "fuchsia-ssh.zbi")
+	_, sshKeyPub, err := genSSHKeys(params.OutputDir)
+	if err != nil {
+		return err
+	}
+
+	sshZBI := filepath.Join(params.OutputDir, "initrd")
 	kernelZBI := filepath.Join(params.KernelDir, "out", arch, "fuchsia.zbi")
-	authorizedKeys := fmt.Sprintf("data/ssh/authorized_keys=%s",
-		filepath.Join(params.KernelDir, ".ssh", "authorized_keys"))
-	if _, err := runSandboxed(time.Minute, params.KernelDir, "out/"+arch+"/host_x64/zbi",
+	authorizedKeys := fmt.Sprintf("data/ssh/authorized_keys=%s", sshKeyPub)
+
+	if _, err := osutil.RunCmd(time.Minute, params.KernelDir, "out/"+arch+"/host_x64/zbi",
```

Comment:
@dvyukov in the ci-fuchsia machine everything in /syzkaller is owned by the root user, so doing `runSandboxed` doesn't work on the output dir. Should we make things in /syzkaller be owned by the syzkaller user?

Reply:
Interesting. What does fail? How?
We have this setup for all OSes on all instances. Build shouldn't touch anything outside of the manager/kernel dir and that dir is owner by syzkaller user. And everything in /syzkaller is world readable/executable, so syzkaller user can run binaries, etc.

Reply:
From the errors I saw on ci-fuchsia, writing as the syzkaller user into params.KernelDir is ok, but writing into params.OutputDir caused a permission error.

In this PR I moved the zbi with the ssh-keys directly to the output dir (initrd), instead of setting the ssh key inside params.KernelDir and then copying the new zbi.  The same happened with creating the SSH Keys and extending the FVM so I used osutil.RunCmd everywhere instead of runSandboxed.

We are OK with using RunCmd, but I just want to make sure we don't have an issue with how syzkaller is setup in ci-fuchsia.

Reply:
Ack.